src/components/application_manager/src/rpc_service_impl.cc

```diff
@@ -118,6 +118,28 @@ bool RPCServiceImpl::ManageMobileCommand(
       return false;
     }
 
+    bool needs_encryption = rpc_protection_mediator_->DoesRPCNeedEncryption(
+        function_id,
+        app,
+        correlation_id,
+        protocol_handler_->IsRPCServiceSecure(connection_key));
+    if ((*message)[strings::params][strings::message_type].asInt() ==
+        MessageType::kRequest) {
+      const bool message_protected =
+          (*message)[strings::params][strings::protection].asBool();
+
+      if (needs_encryption && !message_protected) {
+        const auto response = rpc_protection_mediator_->CreateNegativeResponse(
+            connection_key, function_id, correlation_id);
+        SendMessageToMobile(response);
+        return false;
+      }
+      MessageHelper::PrintSmartObject(*message);
+      if (message_protected && !needs_encryption) {
```

Comment:
correct me if i'm wrong, but as far as i understand this is valid only for responses

Reply:
@mked-luxoft  for notifications also